### ISSUE TITLE
[java] ParseException due to annotation

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#2262]
**Affects PMD Version:** 

6.20.0

**Rule:**

Not applicable

**Description:**

I am getting the following exception when running PMD from the Windows command line.

```
Jan 30, 2020 1:45:22 PM net.sourceforge.pmd.processor.PmdRunnable call
FINE: Processing ...\src\main\java\PMDAnnotationParserBug.java
Jan 30, 2020 1:45:22 PM net.sourceforge.pmd.processor.PmdRunnable addError
FINE: Error while processing file: ...\src\main\java\PMDAnnotationParserBug.java
net.sourceforge.pmd.PMDException: Error while parsing ...\src\main\java\PMDAnnotationParserBug.java
        at net.sourceforge.pmd.SourceCodeProcessor.processSourceCodeWithoutCache(SourceCodeProcessor.java:110)
        at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:89)
        at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:51)
        at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:78)
        at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:24)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: net.sourceforge.pmd.lang.java.ast.ParseException: Encountered " "." ". "" at line 12, column 40.
Was expecting one of:
    <IDENTIFIER> ...
    "@" ...
    "@" ...
    "@" ...
    "@" ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...
    <IDENTIFIER> ...

        at net.sourceforge.pmd.lang.java.ast.JavaParser.generateParseException(JavaParser.java:12229)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.jj_consume_token(JavaParser.java:12113)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.MethodDeclarator(JavaParser.java:1654)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.MethodDeclaration(JavaParser.java:1603)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.ClassOrInterfaceBodyDeclaration(JavaParser.java:1296)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.ClassOrInterfaceBody(JavaParser.java:1251)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.ClassOrInterfaceDeclaration(JavaParser.java:707)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.TypeDeclaration(JavaParser.java:618)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.CompilationUnit(JavaParser.java:352)
        at net.sourceforge.pmd.lang.java.AbstractJavaParser.parse(AbstractJavaParser.java:59)
        at net.sourceforge.pmd.SourceCodeProcessor.parse(SourceCodeProcessor.java:121)
        at net.sourceforge.pmd.SourceCodeProcessor.processSource(SourceCodeProcessor.java:185)
        at net.sourceforge.pmd.SourceCodeProcessor.processSourceCodeWithoutCache(SourceCodeProcessor.java:107)
        ... 10 more
```

**Code Sample demonstrating the issue:**

If I remove `@Nullable` then PMD works fine

```
import org.eclipse.jdt.annotation.NonNullByDefault;
import org.eclipse.jdt.annotation.Nullable;

@NonNullByDefault
public final class PMDAnnotationParserBug
{
   private PMDAnnotationParserBug()
   {
      // no instances
   }

   public static PMDAnnotationParserBug.@Nullable Inner make()   // @Nullable causes the parse failure
   {
      return(new Inner());
   }

   public static class Inner
   {
      // nothing to do
   }
}
```

**Running PMD through:** CLI

Windows 10 version 1903

**Comment by oowekyala** [comment]
Thanks for the report! This is a duplicate of #1367

It's already fixed on the 7.0 development branch, but for now your only choice to avoid this failure is to remove the annotation, or the qualification (import Inner and write `@Nullable Inner`).

**Comment by numeralnathan** [comment]
@oowekyala The reproducing code above is a simplified version of the actual code.  Removing the qualification in the actual code.  I will await for 7.0 to be released.
